From f795be00d2aee9da2ad800cb1d9a54642adcd75f Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Thu, 6 Apr 2023 16:12:54 +0200
Subject: [PATCH 055/105] glamor: Fix build without GBM

The functions glamor_egl_fd_from_pixmap()/glamor_egl_fds_from_pixmap()
are not available without GBM support.

So if GBM is not available or too old, the code would fail to link
trying to find the references to those functions.

Make sure we skip that code when glamor is built without GBM.

Signed-off-by: Olivier Fourdan <ofourdan@redhat.com>
---
 glamor/glamor.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/glamor/glamor.c b/glamor/glamor.c
index 0cbc89ee4..617766c2a 100644
--- a/glamor/glamor.c
+++ b/glamor/glamor.c
@@ -1003,6 +1003,7 @@ _glamor_fds_from_pixmap(ScreenPtr screen, PixmapPtr pixmap, int *fds,
                         uint32_t *strides, uint32_t *offsets,
                         CARD32 *size, uint64_t *modifier)
 {
+#ifdef GLAMOR_HAS_GBM
     glamor_pixmap_private *pixmap_priv = glamor_get_pixmap_private(pixmap);
     glamor_screen_private *glamor_priv =
         glamor_get_screen_private(pixmap->drawable.pScreen);
@@ -1030,6 +1031,7 @@ _glamor_fds_from_pixmap(ScreenPtr screen, PixmapPtr pixmap, int *fds,
     default:
         break;
     }
+#endif /* GLAMOR_HAS_GBM */
     return 0;
 }
 
-- 
2.45.2

